src/mcpdotnet/Server/McpServer.cs

```diff
@@ -36,10 +36,10 @@ internal sealed class McpServer : McpJsonRpcEndpoint, IMcpServer
     /// <param name="loggerFactory">Logger factory to use for logging</param>
     /// <param name="serviceProvider">Optional service provider to use for dependency injection</param>
     /// <exception cref="McpServerException"></exception>
-    public McpServer(IServerTransport transport, McpServerOptions options, ILoggerFactory loggerFactory, IServiceProvider? serviceProvider)
+    public McpServer(ITransport transport, McpServerOptions options, ILoggerFactory loggerFactory, IServiceProvider? serviceProvider)
         : base(transport, loggerFactory)
     {
-        _serverTransport = transport;
+        _serverTransport = transport as IServerTransport;
```

Comment:
If this is only used if it's an `IServerTransport`, why not just make the ctor parameter explicitly an `IServerTransport?`?

Reply:
And perhaps document what `null` transport parameter means.